lib/oli_web/plugs/redirect_if_is_admin.ex

```diff
@@ -0,0 +1,18 @@
+defmodule Oli.Plugs.RedirectIfIsAdmin do
+  import Phoenix.Controller
+  alias OliWeb.Router.Helpers, as: Routes
+
+  def init(_params) do
+  end
+
+  def call(conn, _params) do
+    %{current_author: current_author, current_user: current_user, section: section} = conn.assigns
+
+    if not is_nil(current_author) and is_nil(current_user),
+      do:
+        redirect(conn,
+          to: Routes.page_delivery_path(OliWeb.Endpoint, :index_preview, section.slug)
+        ),
```

Comment:
I'm getting this error when the redirect happens because it keeps going down the other plugs. I think we should pipe this with a [Plug.Conn.halt/1](https://hexdocs.pm/plug/Plug.Conn.html#halt/1)
![image](https://github.com/Simon-Initiative/oli-torus/assets/53446634/3793b4e7-5cc8-49e0-aa3b-a3573579d61d)